docs/src/main/asciidoc/cdi-integration.adoc

```diff
@@ -0,0 +1,597 @@
+////
+This guide is maintained in the main Quarkus repository
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/master/docs/src/main/asciidoc
+////
+= Quarkus - CDI Integration Guide
+
+include::./attributes.adoc[]
+:numbered:
+:sectnums:
+:sectnumlevels: 4
+:toc:
+:toclevels: 2
+
+The ArC container is bootstrapped at build time.
+The downside of this approach is that CDI Portable Extensions cannot be supported.
+Nevertheless, the functionality can be achieved using the Quarkus-specific extensions API.
+
+The container is bootstrapped in multiple phases.
+From a high level perspective these phases go as follows:
+
+1. Initialization
+2. Bean discovery 
+3. Registration of synthetic components
+4. Validation
+
+In the _initialization_ phase the preparatory work is being carried out and custom contexts are registered.
+_Bean discovery_ is then the process where the container analyzes all application classes, identifies beans and wires them all together based on the provided metadata.
+Subsequently, the extensions can register _synthetic components_.
+Attributes of these components are fully contolled by the extensions, i.e. are not derived from an existing class.
+Finally, the _deployment is validated_.
+For example, the container validates every injection point in the application and fails the build if there is no bean that satisfies the given required type and qualifiers.
+
+TIP: You can see more information about the bootstrap by enabling additional logging. Simply run the Maven build with `-X,--debug` and grep lines that contain `io.quarkus.arc`. In the <<cdi-reference.adoc#dev-mode,simplified development mode>>, you can use `quarkus.log.category."io.quarkus.arc.processor".level=DEBUG` and two special endpoints are also registered automatically to provide some basic debug info in the JSON format.
```

Comment:
What's "simplified development mode"? Isn't it just "development mode"?

Reply:
IDK. copy&paste. I'll fix it.